### PR TITLE
new little script to copy fields from art to samp

### PR DESCRIPTION
**Pull request by mayabrandi**
new little script to copy fields from art to samp'

### CODE REVIEW

File: EPPs/copy_field_art2samp.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import sys
+
+
+DESC = """
+"""
+
+
+class CopyUDF():
+    def __init__(self, process, samp_udf, art_udf):
+        self.process = process
```
Comment (sylvinite) [inline EPPs/copy_field_art2samp.py]: Self.process is never actually used

File: EPPs/copy_field_art2samp.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            udf = art.udf.get(self.art_udf)
+            if udf is not None:
+                sample.udf[self.samp_udf] = udf
+                sample.put()
+            else:
+                self.failed_udfs = True
+
+def main(lims, args):
+    process = Process(lims, id = args.pid)
+    CUDF = CopyUDF(process, args.samp_udf, args.art_udf)
```
Comment (sylvinite) [inline EPPs/copy_field_art2samp.py]: All caps is usually reserved for constants

File: EPPs/copy_field_art2samp.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+DESC = """
+"""
+
+
+class CopyUDF():
+    def __init__(self, process, samp_udf, art_udf):
+        self.process = process
+        self.artifacts = [a for a in process.all_outputs() if a.type=='Analyte']
+        self.samp_udf = samp_udf
```
Comment (sylvinite) [inline EPPs/copy_field_art2samp.py]: Considering the two extra letters, reducing self.sample_udf to self.samp_udf seem unnecessary

File: EPPs/copy_field_art2samp.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+def main(lims, args):
+    process = Process(lims, id = args.pid)
+    CUDF = CopyUDF(process, args.samp_udf, args.art_udf)
+    CUDF.copy()
+
+    if CUDF.failed_udfs:
+        sys.exit('failed to copy some udfs')
+    else:
+        print >> sys.stderr, 'UDFs were succsessfully copied!'
```
Comment (sylvinite) [inline EPPs/copy_field_art2samp.py]: Typo in successfully

File: EPPs/copy_field_art2samp.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            else:
+                self.failed_udfs = True
+
+def main(lims, args):
+    process = Process(lims, id = args.pid)
+    CUDF = CopyUDF(process, args.samp_udf, args.art_udf)
+    CUDF.copy()
+
+    if CUDF.failed_udfs:
+        sys.exit('failed to copy some udfs')
```
Comment (sylvinite) [inline EPPs/copy_field_art2samp.py]: Count is probably useful here

File: EPPs/copy_field_art2samp.py (hunk 1 above)
Comment (mayabrandi) [reply EPPs/copy_field_art2samp.py]: 👍

File: EPPs/copy_field_art2samp.py (hunk 2 above)
Comment (mayabrandi) [reply EPPs/copy_field_art2samp.py]: This way of naming class instances is done in all EPPs. I will not change it here and now, but will do a major clean up of the whole repo when there is time.

File: EPPs/copy_field_art2samp.py (hunk 3 above)
Comment (mayabrandi) [reply EPPs/copy_field_art2samp.py]: agree

File: EPPs/copy_field_art2samp.py (hunk 5 above)
Comment (mayabrandi) [reply EPPs/copy_field_art2samp.py]: 👍

**Pull request merged**